pkg/controller/core/clusterqueue_controller.go

```diff
@@ -498,3 +499,17 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus() *kueue.ClusterQueuePendingWorkloadsStatus {
+	pendingWorkloads := make([]kueue.ClusterQueuePendingWorkload, 0)
+	for _, workloads := range r.qManager.GetSnapshots() {
+		pendingWorkloads = append(pendingWorkloads, workloads...)
+	}
+	if len(pendingWorkloads) == 0 {
```

Comment:
This is  not specified in KEP, but I prefer to return nil only when the snapshot is not available (for example before the first calculation). Once available, let's return empty list.